bin/eslint.js

```diff
@@ -76,6 +76,18 @@ function isFileIgnoredByLibrary(file) {
   return ignored;
 }
 
+function prunePathsWithinSymlinks(paths) {
```

Comment:
Sure, this is extracting all of the symlinks from `paths` and filtering out any paths which contain the symlink at the front of the path.

For example, if we have
```
a/       -> not a symlink
a/a.txt  -> not a symlink
b/       -> symlink
b/b.txt  -> not a symlink
```

we'll be left with
```js
["a/", "a/a.txt"]
```